tests/tools/commands/test_repository.py

```diff
@@ -238,6 +250,8 @@ def test_extract_lifecycle(pex, pex_tools_env, tmpdir):
             "--no-pypi",
             "--find-links",
             find_links_url,
+            "--find-links",
+            vendored_pip_dists_dir,
```

Comment:
Why does this test change? Is it a behaviour change?

Reply:
This was a behavior change introduced in #2302, which has vendored Pip lazy installing wheel to support `Pip.spawn_build_wheel` just in time instead of using vendored wheel, which was removed. 

In that change I neglected to do an exhaustive search for tests that used "--no-pypi --find-links ..." and needed to build wheels. Those style tests are susceptible to this issue, but in an order dependent way. If any other test that needed to build a wheel ran before this style "--no-pypi --find-links ..." test, the lazy wheel install would be satisfied and short-circuit.

During this PR work, I happened to hit this case iterating on individual test groups and `rm -rf ~/.pex`.